internal/handlers/pg/msg_createindexes.go

```diff
@@ -102,9 +159,38 @@ func (h *Handler) MsgCreateIndexes(ctx context.Context, msg *wire.OpMsg) (*wire.
 				return err
 			}
 
-			if err = pgdb.CreateIndexIfNotExists(ctx, tx, db, collection, index); err != nil {
+			if index.Name == "" {
+				return commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrCannotCreateIndex,
+					fmt.Sprintf(
+						"Error in specification %s :: caused by :: index name cannot be empty",
+						types.FormatAnyValue(indexDoc),
+					),
+					document.Command(),
+				)
+			}
+
+			if _, ok := duplicateChecker[index.Name]; ok {
+				return commonerrors.NewCommandErrorMsgWithArgument(
+					commonerrors.ErrIndexKeySpecsConflict,
```

Comment:
Looks like it should be `IndexAlreadyExists`.

Reply:
Thanks for catching this. Turns out there are different error code for identical index, same index name and same index id.  See https://github.com/FerretDB/FerretDB/pull/2884/commits/381c58f7b86fb1b208e881cc0a3beea754ec1aad